contentpool.install

```diff
@@ -42,6 +49,71 @@ function contentpool_install() {
   foreach ($source->listAll() as $config_name) {
     $config_storage->write($config_name, $source->read($config_name));
   }
+
+  contentpool_generate_keys();
+  contentpool_enable_cors();
+}
+
+/**
+ * Generates the OAuth Keys.
+ */
+function contentpool_generate_keys() {
+  // Build all the dependencies manually to avoid having to rely on the
+  // container to be ready.
+  $dir_name = 'keys';
+  /** @var \Drupal\simple_oauth\Service\KeyGeneratorService $key_gen */
+  $key_gen = \Drupal::service('simple_oauth.key.generator');
+  /** @var \Drupal\simple_oauth\Service\Filesystem\Filesystem $file_system */
+  $file_system = \Drupal::service('simple_oauth.filesystem');
+
+  $relative_path = DRUPAL_ROOT . '/../' . $dir_name;
+  if (!$file_system->isDirectory($relative_path)) {
+    $file_system->mkdir($relative_path);
+  }
+  $keys_path = $file_system->realpath($relative_path);
+  $pub_filename = sprintf('%s/public.key', $keys_path);
+  $pri_filename = sprintf('%s/private.key', $keys_path);
+
+  if ($file_system->fileExist($pub_filename) && $file_system->fileExist($pri_filename)) {
```

Comment:
It's a bit weird that is done like this during install automatically. What will happen when it's not done?

I think it would be something someone should do manually / semi-automated.  Is there no drush command by simple-oauth to do it?

Let's move it to a drush command (drush 9) and document the call in our docs. I think it would make sense to either read the target directory from config or just make it an argument to the drush call.

Then we can add the call to our init-project.sh script for automated testing.
We could add the drush commend to our distribution or provide as patch to simple oauth + add in the patch.

Reply:
I copied this stuff from contenta_jsonapi, they seem to be fine with it

Reply:
I see - let's move it to a drush commandy anyway. you can generate one with "drush generate", then this is real quick and makes upgrading way easier.